File: src/main/java/com/sport/service/validators/PhoneNumberValidator.java
package com.sport.service.validators;

import com.sport.service.annotations.PhoneNumberValid;
import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

public class PhoneNumberValidator implements ConstraintValidator<PhoneNumberValid, String> {

    private static final String REGEX = "^\\+7\\d{10}$";

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {
        if (value == null || value.isBlank()) {
            return true;
        }
        return value.matches(REGEX);
    }
}

File: src/main/kotlin/com/sport/service/validation/joint_training/PhoneNumberValidator.kt
package com.sport.service.validation.joint_training

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class PhoneNumberValidator : ConstraintValidator<PhoneNumberValid, String> {
    private val regex = Regex("^\\+7\\d{10}$")

    override fun isValid(value: String?, context: ConstraintValidatorContext?): Boolean {
        if (value.isNullOrBlank()) return true
        return regex.matches(value)
    }
}